Add firefox/faq (#9490, Fixes #9491, #9498)

amychurchwell opened the pull request:
## Description
Adds firefox/faq SEO page.
Includes redirect for [new firefox/browsers/quantum page](#9502).

## Issue / Bugzilla link
#9490
#9491
#9498

## Testing
[Final copy](https://docs.google.com/document/d/1LAdUXUovciI1N4xNN1lNGmWv-6kdXVunvM7arrGyK-E/edit?ts=5f74df55)
[Design](https://github.com/mozilla/bedrock/issues/9497#issuecomment-707772063)

amychurchwell commented:
Adding blocked label until Raphael and Chad approve demo.

maxxcrawford commented on bedrock/firefox/templates/firefox/faq.html:
```diff
@@ -0,0 +1,105 @@
+{% from "macros-protocol.html" import hero with context %}
+
+{% block page_title %}{{ ftl('page-title') }}{% endblock %}
+{% block page_desc %} {{ ftl('page-desc') }} {% endblock %}
```
nit: white space

This shows up in the meta description with whitespace at the beginning and end of the entry. 

```suggestion
{% block page_desc %}{{ ftl('page-desc') }}{% endblock %}
```

maxxcrawford commented on l10n/en/firefox/faq.ftl:
```diff
@@ -0,0 +1,100 @@
+page-desc = Whether you searched for a fast browser that protects your privacy, this FAQ is here to answer the most pressing { -brand-name-firefox }-related questions.
+
+firefox-faq = { -brand-name-firefox } FAQ
+whether-you-searched-privacy = Whether you searched for a fast browser that protects your privacy, this FAQ is here to answer the most pressing { -brand-name-firefox }-related questions.
```
Is it okay that this line is a duplicate of L9:`page-desc`? As long as the localization team doesn't have to localize the string twice, this can be ignored. 

cc/ @peiying2

amychurchwell replied:
I will consolidate 👍

maxxcrawford commented on l10n/en/firefox/faq.ftl:
```diff
@@ -0,0 +1,100 @@
+# $url4 (url) - link to https://www.mozilla.org/firefox/linux/
+# $url5 (url) - link to https://www.mozilla.org/firefox/mobile/
+
+you-can-easily = You can easily <a href="{ $url }">download the { -brand-name-firefox } desktop browser</a> here. { -brand-name-firefox } works on <a href="{ $url2 }">{ -brand-name-windows }</a>, <a href="{ $url3 }">{ -brand-name-mac-short }</a> and <a href="{ $url4 }">{ -brand-name-linux}</a> devices, and is also available for <a href="{ $url5 }">{ -brand-name-android } and { -brand-name-ios }</ ...
```
Copy (nit): The link should be "here".

<img width="412" alt="image" src="https://user-images.githubusercontent.com/2692333/96772336-9af34600-13a8-11eb-9388-b4be31916fe7.png">

<img width="640" alt="image" src="https://user-images.githubusercontent.com/2692333/96772341-9c247300-13a8-11eb-9c27-dca630385fc6.png">

maxxcrawford commented on l10n/en/firefox/faq.ftl:
```diff
@@ -0,0 +1,100 @@
+# Variables:
+# $url (url) - link to https://www.mozilla.org/firefox/new/
+
+protecting-your-privacy = Protecting your privacy is our number one priority, and we ensure that installing { -brand-name-firefox } on your devices is completely safe — but always make sure you are downloading from a trusted { -brand-name-mozilla }/{ -brand-name-firefox } site, <a href="{ $url }">like our download page</a>.
```
Copy (nit): The link should be "download page". 

<img width="661" alt="image" src="https://user-images.githubusercontent.com/2692333/96772485-d130c580-13a8-11eb-9635-525b03174fd6.png">


<img width="644" alt="image" src="https://user-images.githubusercontent.com/2692333/96772479-cfff9880-13a8-11eb-8728-48b1bc11edde.png">

maxxcrawford commented on l10n/en/firefox/faq.ftl:
```diff
@@ -0,0 +1,100 @@
+# Variables:
+# $url (url) - link to https://www.mozilla.org/firefox/privacy/products/
+
+not-only-is = Not only is { -brand-name-firefox } safe to use, it also helps keep your data and private information safe. The { -brand-name-firefox-browser } automatically blocks known third party trackers, social media trackers, cryptominers and fingerprinters from collecting your data. Learn more about the <a href="{ $url }">privacy in our products</a>.
```
Copy (nit): The link should be "Learn more about the privacy in our products."

<img width="652" alt="image" src="https://user-images.githubusercontent.com/2692333/96772846-561bdf00-13a9-11eb-9f19-cf07d441f35e.png">

<img width="650" alt="image" src="https://user-images.githubusercontent.com/2692333/96772849-574d0c00-13a9-11eb-9fa1-569a0f6fbbbf.png">

maxxcrawford reviewed (changes requested):
There's a visual issue with the footer: 
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/2692333/96770343-ef48f680-13a5-11eb-8839-174ccccb1f5d.png">

amychurchwell commented on bedrock/firefox/templates/firefox/faq.html:
```diff
@@ -0,0 +1,105 @@
+{% block content %}
+<div class="mzp-l-content">
+
+  {% call hero(
+    title=ftl('firefox-faq'),
+    class='mzp-has-image mzp-t-firefox mzp-t-product-firefox',
+    image_url='img/firefox/browsers/quantum/browser.jpg',
+    include_highres_image=True,
+    include_cta=True,
+    heading_level=1
+  ) %}
```
Thanks Maxx! This comment will be a WONTFIX from me, because the wordmark logo is not the convention we use for [page heros](https://protocol.mozilla.org/patterns/organisms/hero.html#a-firefox-branded-hero-with-image).

amychurchwell commented on bedrock/firefox/templates/firefox/faq.html:
```diff
@@ -0,0 +1,105 @@
+        <p>{{ ftl('firefox-does-not',
+              url='https://fpn.firefox.com/?utm_source=www.mozilla.org-firefox-faq&amp;utm_medium=referral&amp;utm_campaign=seo',
+              url2='https://vpn.mozilla.org/?utm_source=www.mozilla.org-firefox-faq&amp;utm_medium=referral&amp;utm_campaign=seo') }}</p>
+        <small>{{ ftl('related-questions-ip') }}</small>
```
Whoops. That string was missing a `<p>`. Good catch, thank you.

amychurchwell commented:
Thanks for the thoughtful review, @maxxcrawford! Fixed that failing redirect test, adjusted links and copy accordingly, and made that `<li>` scss selector a little more specific to fix the footer.

I also included some copy changes on `quantum.ftl` [per Peiying's comments.](https://github.com/mozilla-l10n/www-l10n/pull/121#pullrequestreview-513969420)

amychurchwell commented:
@maxxcrawford structured data has been added and this has been approved to merge if it looks good to you!
